test/functional/blocknotify.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""Test blocknotify.
+
+Verify that a bitcoind node calls the given command for each block
+"""
+
+import os
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class BlockNotifyTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+
+        self.setup_clean_chain = False
+        self.num_nodes = 1
+
+    def setup_network(self):
+        self.path = os.path.join(self.options.tmpdir, 'blocks.txt')
+
+        # the command will append the notified block hash to file
+        self.extra_args = [['-blocknotify=echo %%s >> %s' % self.path]]
+        super().setup_network()
+
+    def run_test(self):
+        block_count = 10
+        blocks = self.nodes[0].generate(block_count)
+
+        # wait at most 10 seconds for expected file size before reading the content
+        attempts = 0
+        while os.stat(self.path).st_size < (block_count * 65) and attempts < 100:
```

Comment:
Maybe wait until the file is created? I'd really suggest using some sort of wrapper, like the `wait_until` from `mininode`.

```py
wait_until(lambda: os.path.isfile(self.path), timeout=1)
wait_until(lambda: os.stat(self.path).st_size >= (block_count * 65), timeout=10)

Reply:
ping @promag: Are you still working on this?

Reply:
> ping @promag: Are you still working on this?

I'm not seeing any errors when I run the test, but this is probably caused by a race between the start of this loop and the first notification. Could be fixed by adding a `not os.path.exists(self.path)` check here, or adding `with open(self.path, "w"): pass` to `setup_network`.

Reply:
Indeed. The code I posted a few comments above fixed it for me.

c.f.

```py
wait_until(lambda: os.path.isfile(self.path), timeout=1)
wait_until(lambda: os.stat(self.path).st_size >= (block_count * 65), timeout=10)